lib/tile-analyze.js

```diff
@@ -44,102 +45,169 @@ class TileAnalyzeStream extends Transform {
 
 /**
  * Returns stats about an MBTiles file.
+ * Check if the tilestats object has been pre-generated in the mbtiles file.
+ * If so, return that, otherwise generate it with analyzeSourceStream.
  *
  * @param {string} filePath
  * @param {Object} [options]
  * @param {Array<string>} [options.attributes]
+ * @param {boolean} [options.intoMd]
  * @return {Object} The stats.
  */
-module.exports = function(filePath, options) {
-  options = options || {};
-  const stats = createStats();
-  const layerMap = {};
+module.exports = async function tileAnalyze(filePath, options) {
+  const source = await getSource(filePath);
+
+  const info = await getMetadata(source);
+  if (info.tilestats && validator(info.tilestats)) {
+    if (options.intoMd) throw new Error('Tilestats already exists, use without --into-md');
```

Comment:
Could you please make it more descriptive? E.g. "Tilestats already exist. See metadata table, record/row json" or similar. Thanks :)

Reply:
Can be: _Tilestats already exist in json record of metadata table, run without --into-md to display them_

Reply:
Sounds good!